flake.nix

```diff
@@ -0,0 +1,80 @@
+{
+  description = "NixOS systems for internal services";
+
+  inputs = {
+    nixpkgs.url = "github:serokell/nixpkgs";
+    serokell-nix.url = "github:serokell/serokell.nix";
+    flake-compat = {
+      url = "github:edolstra/flake-compat";
+      flake = false;
+    };
+    deploy-rs.url = "github:serokell/deploy-rs";
+  };
+
+  outputs = { self, nixpkgs, serokell-nix, deploy-rs, ... }@inputs:
+    let
+      inherit (nixpkgs.lib) nixosSystem filterAttrs const recursiveUpdate;
+      inherit (builtins) readDir mapAttrs;
+      system = "x86_64-linux";
+      servers = mapAttrs (path: _: import (./servers + "/${path}"))
+        (filterAttrs (_: t: t == "directory") (readDir ./servers));
+      mkSystem = config:
+        nixosSystem {
+          inherit system;
+          modules = [ config ./common.nix ];
+          specialArgs.inputs = inputs;
+        };
+
+      deployChecks =
+        mapAttrs (_: lib: lib.deployChecks self.deploy) deploy-rs.lib;
+
+      terraformFor = pkgs: pkgs.terraform.withPlugins (p: with p; [ aws ]);
+
+      checks = mapAttrs (_: pkgs:
+        let pkgs' = pkgs.extend serokell-nix.overlay;
+        in {
+          trailing-whitespace = pkgs'.build.checkTrailingWhitespace ./.;
+          # FIXME VPC provider is not packaged
+          # terraform = pkgs.runCommand "terraform-check" {
+          #   src = ./terraform;
+          #   buildInputs = [ (terraformFor pkgs) ];
+          # } ''
+          #   cp -r $src ./terraform
+          #   terraform init -backend=false terraform
+          #   terraform validate terraform
+          #   touch $out
+          # '';
+        }) nixpkgs.legacyPackages;
+    in {
+      nixosConfigurations = mapAttrs (const mkSystem) servers;
+
+      nixosModules = import ./modules;
+
+      overlay = import ./packages;
+
+      packages = nixpkgs.lib.genAttrs [ "x86_64-linux" "aarch64-linux" ]
```

Comment:
Did we have an `aarch64-linux` builder?

Reply:
https://github.com/serokell/aquarius-infra/pull/23